services-toolkit/explanation/four-levels-of-service-consumption.hbs.md

```diff
@@ -0,0 +1,69 @@
+# Four Levels of Service Consumption in Tanzu Application Platform
+
+Over time, as Tanzu Application Platform (referred to as "TAP" from hereon in) has evolved and matured, so has the way in which services can be offered and consumed on the platform. In this document we chart the progress of this evolution in terms of four "levels" of service consumption.
+
+## Level 1 - Direct Bindings
+
+When TAP 1.0 was first released way back when, it shipped with support for service bindings, which powers the level 1 concept of direct bindings.
+
+![Diagram shows level 1 of service consumption in TAP](../../images/stk-4-levels-1.png)
+
+Let's imagine that you have deployed an application workload to TAP. As a result you have a Knative service sitting in namespace `foo`. Let's also imagine that there's some other API resource sitting in namespace `foo` which represents a service, perhaps something like a database or a cache. We'll call this resource a service resource. Using a service binding you can bind that service resource onto the Knative service. This has the effect of injecting the credentials for the service resource onto the Knative service, such that ultimately the application workload can then consume it.
+
+This is a relatively straightforward concept to grasp. There are only a few resources at play and there's no unnecessary indirection. In fact, users of TAP are not even directly exposed to the service binding. The service binding is created automatically as part of the supply chain whenever an application workload is configured to refer to a service.
+
+However there are a number of limitations with this sort of setup. The first is that the service resource has to be bindable. That is to say that it must adhere to [Provisioned Service](https://github.com/servicebinding/spec#provisioned-service) as defined by the Service Binding Specification for Kubernetes. While there are _some_ resources out there that do adhere to this specification (primarily resources offered by VMware Tanzu's data services), the overhwelming majority of resources don't.
+
+The second limitation is that all resources have to be in the same namespace, which is ok, but it is a bit limiting.
+
+The third limitation is that the service binding has to know detailed and specific information about the service resource, including its name, namespace and API Group/Version and Kind. This is not a particularly nice or clear separation of concerns as it introduces tight coupling between app teams, who create the application workloads, and ops teams, who create the service resources.
+
+## Level 2 - ResourceClaims
+
+In order to address some of these limitations, to coincide with the release of TAP 1.0 services toolkit shipped with a capability known as ResourceClaims.
+
+![Diagram shows level 2 of service consumption in TAP](../../images/stk-4-levels-2.png)
```

Comment:
**nit:** Most of the diagrams include namespaces. I think they could benefit from a small margin so that the namespace boundaries are more visible. I'm reading your doc in the browser in dark mode and basically can't see the full namespace boxes as they blend into the background.

Reply:
Good catch - will update.